src/utils/useUpdateEffect/useUpdateEffect.ts

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+
+// useUpdateEffect is a useEffect that doesn't fire on initial render
+// for detecting when values change
+// https://stackoverflow.com/a/58217148
+
+const useIsMounted = function useIsMounted() {
+    const isMounted = React.useRef(false);
+
+    React.useEffect(function setIsMounted() {
+        isMounted.current = true;
+
+        return function cleanupSetIsMounted() {
+            isMounted.current = false;
+        };
+    }, []);
+
+    return isMounted;
+};
+
+export const useUpdateEffect = function useUpdateEffect(
```

Comment:
Again, unclear on the same-name function assignment.

Reply:
I get it now. I think it's not necessary to put a comment for every instance this syntax is used as long as there's one near the top explaining it.